vent/coordinator/coordinator.py

```diff
@@ -72,127 +62,146 @@ def __init__(self, sim_mode=False):
-            self.lock.release()
-            for name in [ValueName.PIP,
-                         ValueName.PIP_TIME,
-                         ValueName.PEEP,
-                         ValueName.BREATHS_PER_MINUTE,
-                         ValueName.INSPIRATION_TIME_SEC]:
-                self.lock.acquire()
-                in_control_settings = name not in self.COPY_control_settings
-                self.lock.release()
-                if in_control_settings:
-                    control_setting = self.control_module.get_control(name)
-                    self.lock.acquire()
-                    self.COPY_control_settings[name] = control_setting
-                    self.lock.release()
-
-                self.lock.acquire()
-                disagreed_tentative = name in self.COPY_tentative_control_settings and self.COPY_tentative_control_settings[name] != self.COPY_control_settings[
-                            name]
-                self.lock.release()
-                if disagreed_tentative:
-                    self.lock.acquire()
-                    tentative_control_setting = self.COPY_tentative_control_settings[name]
-                    self.lock.release()
-                    self.control_module.set_control(tentative_control_setting)
-                    self.lock.acquire()
-                    self.COPY_control_settings[name] = self.control_module.get_control(name)
-                    self.lock.release()
-            # sleep 10 ms
-            time.sleep(0.01)
+        """
+        Start the coordinator.
+        This does a soft start (not allocating a process).
+        """
+        self.control_module.start()
+
+    def is_running(self) -> bool:
+        """
+        Test whether the whole system is running
+        """
+        return self.control_module._running
 
     def stop(self):
-        self.stopping.set()
+        """
+        Stop the coordinator.
+        This does a soft stop (not kill a process)
+        """
+        self.control_module.stop()
 
 
 class CoordinatorRemote(CoordinatorBase):
     def __init__(self, sim_mode=False):
         super().__init__(sim_mode=sim_mode)
-        # TODO: pass max_heartbeat_interval
-        self.process_manager = ProcessManager()
-        self.rpc = IPC()
+        # TODO: according to documentation, pass max_heartbeat_interval?
+        self.process_manager = ProcessManager(sim_mode)
+        self.proxy = get_rpc_client()
+        # TODO: make sure the ipc connection is setup. There should be a clever method
+
+    def get_sensors(self) -> Dict[ValueName, SensorValueNew]:
+        sensor_values = pickle.loads(self.proxy.get_sensors().data)
+        res = {
+            ValueName.PIP: SensorValueNew(ValueName.PIP, sensor_values.pip, sensor_values.timestamp,
+                                          sensor_values.loop_counter),
+            ValueName.PEEP: SensorValueNew(ValueName.PEEP, sensor_values.peep, sensor_values.timestamp,
```

Comment:
Maybe the control_module get_sensors() could return in this format so there is not need to convert?

Reply:
@mschottdorf I changed the api of controller.get_sensors(), but then the controller test would fail. Would you like to help me refactor that? The branch is in : https://github.com/CohenLabPrinceton/Ventilator-Dev/tree/refactor_controller_api

Reply:
I have a proposed fix to this that i've been waiting to pull and have commented in several places.

Reply:
basically involves hot hard coding attributes in `SensorValues` https://github.com/CohenLabPrinceton/Ventilator-Dev/blob/d1f96f0823fbee9a4df9ea6dd0b0596b5b393b83/vent/common/message.py#L15